onyo/commands/get.py

```diff
@@ -0,0 +1,158 @@
+        keys: Union[list, None] = None, reverse: bool = False) -> list:
+    """
+    Sort the output of `Repo.get()` by a given list of `keys` or by the path
+    of the `assets` if no `keys` are provided.
+    """
+    if keys:
+        for key in reversed(keys):
+            assets = sorted(
+                assets,
+                key=lambda x: [
+                    int(s) if s.isdigit() else s.lower() for s in
+                    re.split('([0-9]+)', str(x[1][key]))],
+                reverse=reverse)
+    else:
+        assets = sorted(
+            assets,
+            key=lambda x: [
+                int(s) if s.isdigit() else s.lower()
+                for s in re.split('([0-9]+)', str(x[0]))],
+            reverse=reverse)
+
+    return assets
+
+
+def fill_unset(
+        assets: Generator[tuple[Path, dict[str, str]], None, None],
+        keys: list, unset: str = UNSET_VALUE) -> Generator:
+    """
+    If a key is not present for an asset, define it as `unset`.
+    """
+    unset_keys = {key: unset for key in keys}
+    for asset, data in assets:
+        yield asset, unset_keys | data
+
+
+def set_filters(
+        filters: list[str], repo: Repo, rich: bool = False) -> list[Filter]:
+    """Create filters and check if there are no duplicate filter keys"""
+    init_filters = []
+    try:
+        init_filters = [Filter(f, repo=repo) for f in filters]
+    except OnyoInvalidFilterError as exc:
+        if rich:
+            console = Console(stderr=True)
+            console.print(f'[red]FAILED[/red] {exc}')
+        else:
+            print(exc, file=sys.stderr)
+        sys.exit(1)
+
+    # ensure there are no duplicate filter keys
+    duplicates = [
+        x for x, i in Counter([f.key for f in init_filters]).items() if i > 1]
+    if duplicates:
+        if rich:
+            console = Console(stderr=True)
+            console.print(
+                f'[red]FAILED[/red] Duplicate filter keys: {duplicates}')
+        else:
+            print(f'Duplicate filter keys: {duplicates}', file=sys.stderr)
+        sys.exit(1)
+    return init_filters
+
+
+def sanitize_keys(k: list[str], defaults: list) -> list[str]:
+    """
+    Remove duplicates from k while preserving key order and return default
+    (pseudo) keys if k is empty
+    """
+    seen = set()
+    k = [x for x in k if not (x in seen or seen.add(x))]
+    return k if k else defaults
+
+
+def get(args: argparse.Namespace, opdir: str):
+    """
+    Display the requested `key`(s) in tabular form for matching assets.
+    """
+    if args.sort_ascending and args.sort_descending:
+        msg = (
+            '--sort-ascending (-s) and --sort-descending (-S) cannot be used '
+            'together')
+        if args.machine_readable:
+            print(msg, file=sys.stderr)
+        else:
+            console = Console(stderr=True)
+            console.print(f'[red]FAILED[/red] {msg}')
+        sys.exit(1)
+
+    repo = None
+    try:
+        repo = Repo(opdir)
```

Comment:
Done, I temporarily removed this because it caused issues when running `onyo get` in sub-directories. `onyo get` works fine in sub-directories without it. I'll be looking into `fsck` in another PR to address this issue. The error there is caused by the assumption that `git ls-files` always performs from root, whereas it actually performs from cwd.